compiler+runtime/include/cpp/jank/runtime/obj/ratio.hpp

```diff
@@ -0,0 +1,146 @@
+#pragma once
+
+namespace jank::runtime
+{
+  template <>
+  struct static_object<object_type::ratio_data> : gc
+  {
+    static constexpr native_bool pointer_free{ true };
+    static_object() = default;
+    static_object(static_object &&) = default;
+    static_object(static_object const &) = default;
+    static_object(native_integer const numerator, native_integer const denominator);
+
+    native_real to_real() const;
+    void simplify();
+
+    native_integer numerator{};
+    native_integer denominator{};
+  };
```

Comment:
I don't think you want a new jank object type for `ratio_data`. You didn't add it to `erasure.hpp` for the same reason. We don't ever want a `ratio_data` on its own. It's just a detail of a `ratio`. You also didn't implement any of the `objectable` interface, so this isn't a proper jank object anyway. This should just be a struct with none of the other stuff involved.

```cpp
namespace obj
{
  struct ratio_data
  {
    ratio_data(native_integer numerator, native_integer denominator);

    native_real to_real() const;

    native_integer numerator{};
    native_integer denominator{};
  };
}
```

There is also no need for a `simplify` function. Just do that in the ctor.

Reply:
i've made a large change to the use of `ratio_data`, changing it to a non-jank object.